src/apps/tools/OswAppCalculator.cpp

```diff
@@ -0,0 +1,383 @@
+/*
+    a calculator for addition, subtraction, multiplication, division, square roots, powers, num1 * 10 ^ num2
+    with postitive and negativ numbers + generating random numbers
+    press btn2 + btn3 to move the cursor a position back
+*/
+
+#include "./apps/tools/OswAppCalculator.h"
+
+#include "config_defaults.h"
+#include "gfx_util.h"
+#include "osw_app.h"
+#include "osw_hal.h"
+#include "osw_ui.h"
+
+#include <osw_config_keys.h>
+
+int num1[] = {0, 0, 0, 0, 0, 0};
+int num2[] = {0, 0, 0, 0, 0, 0};
+
+
+void OswAppCalculator::setup() {}
+
+void OswAppCalculator::loop() {
+    OswHal* hal = OswHal::getInstance();
+
+    static int8_t position = 0;
+    static uint8_t iNum = 0;
+    static String resultToString = "result";
+    static char signOfNum1 = '+';
+    static String mathOperation = "+";
+    static char signOfNum2 = '+';
```

Comment:
Le's move those into the class itself as well as the vars inside this source header. This will allow us to reset them as needed by class destruction.